exercises/binary-search/test/test_binary_search.c

```diff
@@ -9,89 +9,98 @@ void tearDown(void)
 {
    TEST_IGNORE();               // delete this line to run test
    int arr[] = { 1, 3, 4, 6, 8, 9, 11 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(&arr[3] == binary_search(6, arr, length));
 }
 
-static void test_value_at_beginning(void)
+static void test_finds_a_value_at_the_beginning_of_an_array(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 4, 6, 8, 9, 11 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(&arr[0] == binary_search(1, arr, length));
 }
 
-static void test_value_at_end(void)
+static void test_finds_a_value_at_the_end_of_an_array(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 4, 6, 8, 9, 11 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(&arr[6] == binary_search(11, arr, length));
 }
 
-static void test_find_value_with_odd_length(void)
+static void test_finds_a_value_in_an_array_of_odd_length(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 634 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(&arr[9] == binary_search(144, arr, length));
 }
 
-static void test_find_value_with_even_length(void)
+static void test_finds_a_value_in_an_array_of_even_length(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(&arr[5] == binary_search(21, arr, length));
 }
 
-static void test_identify_missing_value(void)
+static void test_identifies_that_a_value_is_not_included_in_the_array(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 4, 6, 8, 9, 11 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(NULL == binary_search(7, arr, length));
 }
 
-static void test_value_smaller_than_everything(void)
+static void
+test_a_value_smaller_than_the_arrays_smallest_value_is_not_found(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 4, 6, 8, 9, 11 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(NULL == binary_search(0, arr, length));
 }
 
-static void test_value_larger_than_everything(void)
+static void test_a_value_larger_than_the_arrays_largest_value_is_not_found(void)
 {
    TEST_IGNORE();
    int arr[] = { 1, 3, 4, 6, 8, 9, 11 };
    size_t length = sizeof(arr) / sizeof(arr[0]);
    TEST_ASSERT(NULL == binary_search(13, arr, length));
 }
 
-static void test_empty_array(void)
+static void test_nothing_is_found_in_an_empty_array(void)
 {
    TEST_IGNORE();
-   int *arr = NULL;
+   int arr[] = { 1 };
    size_t length = 0;
    TEST_ASSERT(NULL == binary_search(1, arr, length));
 }
 
-static void test_zero_length_array(void)
+static void test_nothing_is_found_when_the_left_and_right_bounds_cross(void)
 {
    TEST_IGNORE();
-   int arr[] = { 1 };
+   int arr[] = { 1, 2 };
+   size_t length = sizeof(arr) / sizeof(arr[0]);
+   TEST_ASSERT(NULL == binary_search(0, arr, length))
+}
+
+static void test_nothing_is_found_if_array_is_null(void)
```

Comment:
This test is additional to the spec.
I am in two minds if to keep it in or not.
On one hand it seems a logical test for idiomatic C based languages, on the other hand it diverges from the spec.

This question comes about because of the interpretation of the semantics of the spec for the (specified) test `test_nothing_is_found_in_an_empty_array()`.
As implemented here, the aforementioned test checks what it says, an empty array. That is, an array that exists, but that has nothing in it.
Whereas this test in question checks the case that is more idiomatic to C specifically, i.e. the array does not exist.

There are three options
1. It's not important to check every C idiomatic thing on every exercise, so remove this test and stick to a strict interpretation of the specification
2. Remove this test, but change `test_nothing_is_found_in_an_empty_array()` to actually check for no array, rather than an empty array (as this would be a more likely implementation for "empty array" in C), resulting in tests that follow the specification, but with a slightly looser interpretation.
3. Keep both tests as they are - ( i.e. the strict specification semantics + our extra test )

Reply:
I lean toward 1). For something like Exercism (which definitely isn't production code), having a ton of checks on every exercise just gets in the way of the important parts. I'm happy with including idiomatic checks in some exercises and letting the "meat" of the exercises take center stage in the general case.